From 8631704be06e00cb7a7a415f13811ca4ece99e8b Mon Sep 17 00:00:00 2001
Message-Id: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
From: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
Date: Thu, 10 Sep 2015 12:07:21 +0530
Subject: [PATCH 01/10] REVERTME [VPG]: drm/i915: Allowing enumeration of
 multiple external devices

Mulitple external display enumeration was restricted earlier due
to support for audio is available only for one external display.
This is removed in this patch as audio will now support dynamic
switching of displays.

REVERTME because lpe audio related changes are not upstream

Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-10172
Change-Id: I472f4679839abe749fd8ac85a1a759ce08ee3d90
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Signed-off-by: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 66 ++++--------------------------------
 1 file changed, 7 insertions(+), 59 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 18ac887..5e758d8 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -13325,9 +13325,6 @@ static bool intel_crt_present(struct drm_device *dev)
 static void intel_setup_outputs_vbt(struct drm_device *dev)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
-	int hdmi_ports = 0;
-	int dp_ports = 0;
-	int port = 0;
 	int i;
 
 	for (i = 0; i < dev_priv->vbt.child_dev_num; i++) {
@@ -13347,7 +13344,7 @@ static void intel_setup_outputs_vbt(struct drm_device *dev)
 		case DVO_PORT_DPC:
 		case DVO_PORT_DPD:
 			if (devtype & DEVICE_TYPE_eDP_BITS) {
-				port = dvo_port - DVO_PORT_CRT;
+				int port = dvo_port - DVO_PORT_CRT;
 				intel_dp_init(dev, VLV_DISPLAY_BASE
 					+ DDI_BUF_CTL(port), port);
 			}
@@ -13357,13 +13354,16 @@ static void intel_setup_outputs_vbt(struct drm_device *dev)
 		case DVO_PORT_HDMID:
 			if (devtype == DEVICE_TYPE_DP_HDMI_DVI ||
 						devtype == DEVICE_TYPE_HDMI) {
-				/* Delay HDMI init, save the port */
-				hdmi_ports |= (1 << dvo_port);
+				int hdmi_reg = VLV_DISPLAY_BASE
+						+ PORT_ADDR(dvo_port);
+				intel_hdmi_init(dev, hdmi_reg, dvo_port);
 			}
 
 			if (devtype == DEVICE_TYPE_DP_HDMI_DVI ||
 						devtype == DEVICE_TYPE_DP) {
-				dp_ports |= (1 << dvo_port);
+				intel_dp_init(dev, VLV_DISPLAY_BASE
+						+ DDI_BUF_CTL(dvo_port),
+							dvo_port);
 			}
 			break;
 		default:
@@ -13372,58 +13372,6 @@ static void intel_setup_outputs_vbt(struct drm_device *dev)
 
 		}
 	}
-
-	/*
-	* Current audio driver can support only one
-	* instance of HDMI. Dual HDMI enumeration causes
-	* panic during bootup. To handle this, for now
-	* add a preference order for enumeration of displays
-	* in driver: HDMI port D over port B over port C
-	* If the device has single HDMI enemurated, driver
-	* picks that port, but in case of multiple HDMI, it
-	* selects PORT D
-	*/
-	if (hdmi_ports) {
-		int hdmi_reg = 0;
-		if (hdmi_ports & (1 << DVO_PORT_HDMID)) {
-			port = DVO_PORT_HDMID;
-			hdmi_reg = VLV_DISPLAY_BASE
-				+ PORT_ADDR(port);
-		} else if (hdmi_ports & (1 << DVO_PORT_HDMIB)) {
-			port = DVO_PORT_HDMIB;
-			hdmi_reg = VLV_DISPLAY_BASE
-				+ PORT_ADDR(port);
-		} else if (hdmi_ports & (1 << DVO_PORT_HDMIC)) {
-			port = DVO_PORT_HDMIC;
-			hdmi_reg = VLV_DISPLAY_BASE
-				+ PORT_ADDR(port);
-		}
-
-		/*
-		 * Since audio is supported only on one external
-		 * panel, either HDMI or DP, we are giving
-		 * preference to HDMI over DP.
-		 */
-		dev_priv->support_dp_audio = false;
-
-		intel_hdmi_init(dev, hdmi_reg, port);
-		DRM_DEBUG_DRIVER("HDMI port=%c\n",
-			port_name(port));
-	} else {
-		dev_priv->support_dp_audio = true;
-	}
-
-	if (dp_ports) {
-		if (dp_ports & (1 << DVO_PORT_HDMIB))
-			port = DVO_PORT_HDMIB;
-		else if (dp_ports & (1 << DVO_PORT_HDMIC))
-			port = DVO_PORT_HDMIC;
-		else if (dp_ports & (1 << DVO_PORT_HDMID))
-			port = DVO_PORT_HDMID;
-
-		intel_dp_init(dev, VLV_DISPLAY_BASE + DDI_BUF_CTL(port), port);
-		DRM_DEBUG_KMS("DP on port=%c\n", port_name(port));
-	}
 }
 
 static void intel_setup_outputs(struct drm_device *dev)
-- 
1.9.1

